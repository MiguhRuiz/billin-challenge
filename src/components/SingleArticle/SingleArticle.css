.SingleArticle {
    margin: 3em;
}

.SingleArticle-Name {
    text-align: center;
}

.SingleArticle-Author {
    margin-bottom: 3em;
    text-align: center;
}

.SingleArticle-Content {
    font-size: 1.2em;
    line-height: 1.5em;
    margin-bottom: 4em;
}

.SingleArticle-Tags {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.SingleArticle-Tags-Tags {
    font-weight: 800;
    margin-left: 2em;
}