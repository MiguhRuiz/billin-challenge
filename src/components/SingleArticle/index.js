import React from 'react'
import './SingleArticle.css'

function SingleArticle(props) {
    return(
        <div className="App">
            <article className="SingleArticle">
                <div className="SingleArticle-Menu">
                    <h1 className="SingleArticle-Name">{props.title}</h1>
                    <button className="SingleArticle-Update fa fa-pencil" onClick={props.updateArticle}/>
                    <button className="SingleArticle-Delete fa fa-trash" onClick={props.deleteArticle}/>
                </div>
                <div className="SingleArticle-Author">
                    <span className="SingleArticle-Author-Detail">Por:</span>
                    <span className="SingleArticle-Author-Name">{props.author}</span>
                </div>
                <div className="SingleArticle-Content">
                    {props.content}
                </div>
                <div className="SingleArticle-Tags">
                    <div className="SingleArticle-Tags-Detail">Este artículo está clasificado en:</div>
                    <div className="SingleArticle-Tags-Tags">{props.tags}</div>
                </div>
            </article>
        </div>
    )
}

export default SingleArticle