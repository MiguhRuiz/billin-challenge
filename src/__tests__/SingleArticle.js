import React from 'react'
import { shallow } from 'enzyme'

import SingleArticle from '../components/SingleArticle'

test('It should render a full article', () => {
    const View = shallow(
        <SingleArticle/>
    )

    expect(View).toMatchSnapshot()
})