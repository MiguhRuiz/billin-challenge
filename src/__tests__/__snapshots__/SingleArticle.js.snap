exports[`test It should render a full article 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App">
    <article
        className="SingleArticle">
        <div
            className="SingleArticle-Menu">
            <h1
                className="SingleArticle-Name" />
            <button
                className="SingleArticle-Update fa fa-pencil"
                onClick={undefined} />
            <button
                className="SingleArticle-Delete fa fa-trash"
                onClick={undefined} />
        </div>
        <div
            className="SingleArticle-Author">
            <span
                className="SingleArticle-Author-Detail">
                Por:
            </span>
            <span
                className="SingleArticle-Author-Name" />
        </div>
        <div
            className="SingleArticle-Content" />
        <div
            className="SingleArticle-Tags">
            <div
                className="SingleArticle-Tags-Detail">
                Este artículo está clasificado en:
            </div>
            <div
                className="SingleArticle-Tags-Tags" />
        </div>
    </article>
</div>,
  "nodes": Array [
    <div
      className="App">
      <article
            className="SingleArticle">
            <div
                  className="SingleArticle-Menu">
                  <h1
                        className="SingleArticle-Name" />
                  <button
                        className="SingleArticle-Update fa fa-pencil"
                        onClick={undefined} />
                  <button
                        className="SingleArticle-Delete fa fa-trash"
                        onClick={undefined} />
            </div>
            <div
                  className="SingleArticle-Author">
                  <span
                        className="SingleArticle-Author-Detail">
                        Por:
                  </span>
                  <span
                        className="SingleArticle-Author-Name" />
            </div>
            <div
                  className="SingleArticle-Content" />
            <div
                  className="SingleArticle-Tags">
                  <div
                        className="SingleArticle-Tags-Detail">
                        Este artículo está clasificado en:
                  </div>
                  <div
                        className="SingleArticle-Tags-Tags" />
            </div>
      </article>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SingleArticle />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="App">
          <article
                    className="SingleArticle">
                    <div
                              className="SingleArticle-Menu">
                              <h1
                                        className="SingleArticle-Name" />
                              <button
                                        className="SingleArticle-Update fa fa-pencil"
                                        onClick={undefined} />
                              <button
                                        className="SingleArticle-Delete fa fa-trash"
                                        onClick={undefined} />
                    </div>
                    <div
                              className="SingleArticle-Author">
                              <span
                                        className="SingleArticle-Author-Detail">
                                        Por:
                              </span>
                              <span
                                        className="SingleArticle-Author-Name" />
                    </div>
                    <div
                              className="SingleArticle-Content" />
                    <div
                              className="SingleArticle-Tags">
                              <div
                                        className="SingleArticle-Tags-Detail">
                                        Este artículo está clasificado en:
                              </div>
                              <div
                                        className="SingleArticle-Tags-Tags" />
                    </div>
          </article>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="App">
          <article
                    className="SingleArticle">
                    <div
                              className="SingleArticle-Menu">
                              <h1
                                        className="SingleArticle-Name" />
                              <button
                                        className="SingleArticle-Update fa fa-pencil"
                                        onClick={undefined} />
                              <button
                                        className="SingleArticle-Delete fa fa-trash"
                                        onClick={undefined} />
                    </div>
                    <div
                              className="SingleArticle-Author">
                              <span
                                        className="SingleArticle-Author-Detail">
                                        Por:
                              </span>
                              <span
                                        className="SingleArticle-Author-Name" />
                    </div>
                    <div
                              className="SingleArticle-Content" />
                    <div
                              className="SingleArticle-Tags">
                              <div
                                        className="SingleArticle-Tags-Detail">
                                        Este artículo está clasificado en:
                              </div>
                              <div
                                        className="SingleArticle-Tags-Tags" />
                    </div>
          </article>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SingleArticle />,
}
`;
